manifest/kcp.yaml

```diff
@@ -0,0 +1,151 @@
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: kcp
+spec:
+  secretName: kcp-secret
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  subject:
+    organizations:
+      - redhat
+  isCA: false
+  privateKey:
+    algorithm: RSA
+    encoding: PKCS1
+    size: 2048
+  usages:
+    - server auth
+    - client auth
+  dnsNames:
+    - kcp
+    - "*openshiftapps.com"
+  ipAddresses:
+    - 0.0.0.0
+  issuerRef:
+    name: kcp
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: kcp
+spec:
+  ports:
+    - protocol: TCP
+      name: kcp
+      port: 6443
+      targetPort: 6443
+    - protocol: TCP
+      name: virtual-workspaces
+      port: 6444
+      targetPort: 6444
+  selector:
+    app: kcp
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kcp
+  labels:
+    app: kcp
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kcp
+  template:
+    metadata:
+      labels:
+        app: kcp
+    spec:
+      containers:
+      - name: kcp
+        image: ghcr.io/kcp-dev/kcp:latest
+        ports:
+        - containerPort: 6443
+        command:
+        - /kcp
+        args:
+        - start
+        - --run-controllers
+        - --auto-publish-apis
+        - --push-mode
+        - --etcd-servers=https://etcd:2379
+        - --etcd-keyfile=/etc/etcd/tls/server/tls.key
+        - --etcd-certfile=/etc/etcd/tls/server/tls.crt
+        - --etcd-cafile=/etc/etcd/tls/server/ca.crt
+        - --tls-private-key-file=/etc/kcp/tls/server/tls.key
+        - --tls-cert-file=/etc/kcp/tls/server/tls.crt
```

Comment:
while technically possible, can we split the two serving certs of kcp and virtual-workspaces and use a common CA ? (can be follow-up). Using the same for both looks like a non-production shortcut.

Reply:
Also this means that we have to put the ca.crt, not the tls.crt into users' and admin's kubeconfig.

Reply:
I've updated the PR to have the virtual workspaces container use a different cert than kcp.